src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorExtensionInitializer.cs

```diff
@@ -72,12 +72,13 @@ public static void Register(RazorProjectEngineBuilder builder)
                 builder.Features.Add(new TrimWhitespacePass());
             }
             builder.Features.Add(new ComponentDocumentClassifierPass());
+            builder.Features.Add(new ComponentDocumentRewritePass());
+            builder.Features.Add(new ScriptTagPass());
             builder.Features.Add(new ComplexAttributeContentPass());
-            builder.Features.Add(new BindLoweringPass());
-            builder.Features.Add(new EventHandlerLoweringPass());
             builder.Features.Add(new ComponentLoweringPass());
+            builder.Features.Add(new EventHandlerLoweringPass());
             builder.Features.Add(new RefLoweringPass());
-            builder.Features.Add(new OrphanTagHelperLoweringPass());
+            builder.Features.Add(new BindLoweringPass());
```

Comment:
These passes did not match the stated ordering. This has been cleaned up. 

Now the component pass runs pretty early, and makes the decision of whether the node is an element or component. It leaves behind 'tag helper property' nodes for the other 3 passes to handle. The expectation is that each 'tag helper property' must belong to either of these 4 later passes, and should never make it to the code generation phase without being replaced.